### ISSUE TITLE
Editor Rulers do not account for Tab display size

### ISSUE THREAD

**Issue by InSuperposition** [microsoft/vscode#228215]
Summary:
When working in Visual Studio Code (VS Code), the editor rulers are visually misaligned with code indentation when using tabs. The rulers assume that every tab is displayed as a fixed number of spaces, but this does not dynamically account for the display width set for tabs. As a result, when tabs are displayed at a size different from the default, the rulers become inaccurate.

Expected Behavior:
The rulers should adjust according to the tab display size. When the tab width is set to 2 spaces, rulers should account for each tab character as occupying 2 columns in the editor, ensuring that the visual representation of the code matches the ruler guide.

Actual Behavior:
The rulers treat each tab as a fixed number of spaces, regardless of the user-configured tab width. This leads to visual misalignment between the ruler and the code when working with non-standard tab sizes. In images below the rulers do not move.

Tabs size 2 spaces:
<img width="1021" alt="Image" src="https://github.com/user-attachments/assets/5415746d-e099-431e-900d-78d85924838e" />

Tabs size 8 spaces:
<img width="1019" alt="Image" src="https://github.com/user-attachments/assets/8806db7b-6096-4f85-bb1d-dd8f14810ae7" />

**Comment by hediet** [comment]
Can you please share a screenshot? Thanks!

**Comment by InSuperposition** [comment]
@hediet I added images to original comment. Thanks.

**Comment by hediet** [comment]
But that would mean that the ruler is not a straight line anymore, correct?

**Comment by InSuperposition** [comment]
No, though I misunderstood the behavior of the rulers initially. I setup a small example using a tab with a size of 8 and a ruler at 10 columns (initially I thought rulers were defined by character, not column). 

Notice the yellow cursor's placement in the 1st image. It only 3 key strokes to reach the ruler from beginning of the line, which is to be expected, IMHO.

I assumed the rulers would be based on the actual characters count as character count is preserved in the file, not on column count which is a merely a visual representation and only in the VS codes settings.

<img width="377" alt="Image" src="https://github.com/user-attachments/assets/46226a72-acec-4c59-9f99-1cef1b3733b1" />

Column count is at 11.
<img width="168" alt="Image" src="https://github.com/user-attachments/assets/3fa27952-01ad-4d7d-94c8-c2e22ff716b2" />

Character count is 4.
<img width="510" alt="Image" src="https://github.com/user-attachments/assets/a7f5fbe2-a75f-4c83-88c6-036e5f10a60a" />

**Comment by hediet** [comment]
Closing as it works as intended. Rulers use columns, which are independent of the tab size.
